src/ResourceManager/Blueprint/Commands.Blueprint/Cmdlets/GetAzureRMBlueprint.cs

```
@@ -0,0 +1,273 @@
+                        HandleLatestPublishedParameterSet();
+                        break;
+                }
+            }
+            catch (Exception ex)
+            {
+                WriteExceptionError(ex);
+            }
+        }
+        #endregion Cmdlet Overrides
+
+        #region Private Methods
+        /// <summary>
+        /// Handle the "Default" parameter set.
+        /// </summary>
+        /// <remarks>
+        /// The "Default" parameter set fetches either all blueprints or named blueprints.
+        /// Both published and draft blueprints are retrieved.
+        /// </remarks>
+        private void HandleDefaultParameterSet()
+        {
+            if (ManagementGroupName == null || ManagementGroupName.Length > 1)
+            {
+                ProcessBlueprintsFromAllManagementGroups();
+            }
+            else
+            {
+                if (Name == null)
+                {
+                    ProcessBlueprintsFromManagementGroup(ManagementGroupName[0]);
+                }
+                else
+                {
+                    ProcessNamedBlueprints(ManagementGroupName[0]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Handle the "ByVersion" parameter set.
+        /// </summary>
+        /// <remarks>
+        /// The "ByVersion" parameter set filters published blueprints by version name.
+        /// The "Name" parameter is required for this parameter set.
+        /// </remarks>
+        private void HandleByVersionParameterSet()
+        {
+            foreach (var name in Name)
+            {
+                try
+                {
+                    var bp = BlueprintClient.GetPublishedBlueprint(ManagementGroupName[0], name, VersionName);
+                    WriteObject(bp);
+                }
+                catch (Exception ex)
+                {
+                    if (Name.Length == 1)
+                        WriteExceptionError(ex);
```

Don't swallow exceptions here. This will silently do nothing if Name.Length is anything but 1.
Also, please use braces.

---

Ignoring errors unless a single blueprint Name is given is the requested behavior.

I'll add the braces.

---

OK, then that's the design and it's an unusual anti-pattern. Many people reading it will be suspicious, so it would help a lot to comment each catch block that does it and explain the design/reasoning.

---

Added braces and comment.